editoast/src/views/infra/mod.rs

```diff
@@ -350,23 +350,51 @@ async fn get_speed_limit_tags(
     ))
 }
 
+#[derive(Debug, Clone, Deserialize)]
+struct GetVoltagesQueryParams {
+    #[serde(default)]
+    add_modes: bool,
+}
+
 /// Returns the set of voltages for a given infra
 #[get("/voltages")]
-async fn get_voltages(infra: Path<i64>, db_pool: Data<DbPool>) -> Result<Json<Vec<f64>>> {
+async fn get_voltages(
+    infra: Path<i64>,
+    param: Query<GetVoltagesQueryParams>,
+    db_pool: Data<DbPool>,
+) -> Result<Json<Vec<String>>> {
     let infra = infra.into_inner();
+    let add_modes = param.into_inner().add_modes;
     let voltages: Vec<Voltage> = block::<_, Result<_>>(move || {
         let mut conn = db_pool.get()?;
-        match sql_query(
-            "SELECT DISTINCT ((data->'voltage')->>0)::float AS voltage
+        let query = "SELECT DISTINCT ((data->'voltage')->>0) AS voltage
                 FROM osrd_infra_catenarymodel
-                WHERE infra_id = $1
-                ORDER BY voltage",
-        )
-        .bind::<BigInt, _>(infra)
-        .load(&mut conn)
-        {
-            Ok(voltages) => Ok(voltages),
-            Err(err) => Err(err.into()),
+                WHERE infra_id = $1";
+        if !add_modes {
+            match sql_query(String::from(query) + " ORDER BY voltage")
+                .bind::<BigInt, _>(infra)
+                .load(&mut conn)
+            {
+                Ok(voltages) => Ok(voltages),
+                Err(err) => Err(err.into()),
+            }
```

Comment:
```suggestion
            query = String::from(query) + " ORDER BY voltage"
```
the match sql_query can be be done outside the if ... else ... loop